src/test/java/com/training/services/tests/ProductsTest.java

```diff
@@ -83,19 +83,23 @@ public void testShorexProductTypeValidate() {
      */
     @Test
     public void testShorexOfferingDateValidate() throws ParseException {
-        response = new RestEngine().getResponseGet(baseURL, headerMap, queryParam)
-                .as(Response.class);
+        response = new RestEngine().getResponseGet(baseURL, headerMap, queryParam).as(Response.class);
 
         Map<String, Integer> dayMonthDaysInMonthFromLink = getDayMonthDaysInMonth(map.get("sailingIdParamValue").split("AL")[1]);
 
         List<Products> products = response.getPayload().getProducts();
-        List<Offering> offerings;
 
-        for (int index = 0; index < products.size(); index++) {
-            offerings = products.get(index).getOffering();
-            for (int getDateOffering = 0; getDateOffering < offerings.size(); getDateOffering++) {
+        IntStream.range(0, products.size()).forEach(index -> {
+            List<Offering> offerings = products.get(index).getOffering();
 
-                Map<String, Integer> dayMonthDaysInMonth = getDayMonthDaysInMonth(offerings.get(getDateOffering).getOfferingDate());
+            IntStream.range(0, offerings.size()).forEach(getDateOffering -> {
+                Map<String, Integer> dayMonthDaysInMonth = null;
+
+                try {
+                    dayMonthDaysInMonth = getDayMonthDaysInMonth(offerings.get(getDateOffering).getOfferingDate());
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
```

Comment:
Do we really need index in this?
Why cant we use - response.getPayload().getProducts().forEach()?